src/treeview.ts

```diff
@@ -0,0 +1,186 @@
+import * as path from "path";
+import * as fs from "fs";
+import { LanguageClient, Disposable } from "vscode-languageclient";
+import {
+  TreeDataProvider,
+  TreeItem,
+  Event,
+  EventEmitter,
+  TreeItemCollapsibleState,
+  window,
+  OutputChannel,
+  Uri,
+  TreeView,
+  ExtensionContext
+} from "vscode";
+import {
+  TreeViewNode,
+  MetalsTreeViews,
+  MetalsTreeViewChildren,
+  MetalsTreeViewDidChange,
+  MetalsTreeViewVisibilityDidChange,
+  MetalsRevealTreeView,
+  MetalsTreeViewParent
+} from "./protocol";
+
+export function startTreeView(
+  client: LanguageClient,
+  out: OutputChannel,
+  context: ExtensionContext,
+  packageJson: any
```

Comment:
same here, it would be good to have some level of validation beforehand, so that we don't pass `any` values around.

Reply:
I moved the `const viewIds= ...` logic to `extension.ts` and changed the param to be `viewIds: string[]`. I agree that passing `any` around is bad.